velox/docs/velox-in-10-min.rst

```diff
@@ -0,0 +1,321 @@
+===================
+Velox in 10 minutes
+===================
+
+This is a quick introduction into Velox, the new C++ vectorized database
+acceleration library aimed at optimizing query engines and data processing
+systems. This tutorial was inspired by `TorchArrow in 10 minutes
+<https://github.com/facebookresearch/torcharrow/blob/main/tutorial/tutorial.ipynb>`_
+which itself was based on a `10 minutes to pandas tutorial <https://pandas.pydata.org/docs/user_guide/10min.html>`_.
+
+We are going to briefly touch all major components starting with data
+representation, followed by expression evaluation, followed by aggregations,
+sorting, joins and accessing data via connectors.
+
+Note: Velox doesn't provide SQL interface, but in this tutorial to keep
+things simple we are going to use SQL and utilities that were designed
+to simplify test development. These utilities use `DuckDB <https://duckdb.org/>`_
+to parse SQL.
+
+The code used in this tutorial is available in `velox/exec/tests/VeloxIn10MinDemo.cpp <https://github.com/facebookincubator/velox/blob/main/velox/exec/tests/VeloxIn10MinDemo.cpp>`_.
+
+Let’s get started.
+
+Vectors
+-------
+
+Velox uses columnar representation of the data similar to Arrow called vectors.
+Let’s create two vectors to store 64-bit integers.
+
+.. code-block:: c++
+
+    auto a = makeFlatVector<int64_t>({0, 1, 2, 3, 4, 5});
+    auto b = makeFlatVector<int64_t>({0, 5, 10, 15, 20, 25});
+
+And let’s print out the contents of these vectors:
+
+.. code-block:: c++
+
+    auto data = makeRowVector({"a", "b"}, {a, b});
+
+    std::cout << data->toString() << std::endl;
+    std::cout << data->toString(0, data->size()) << std::endl;
+
+.. code-block::
+
+    > [ROW ROW<a:BIGINT,b:BIGINT>: 6 elements, no nulls]
+
+    0: {0, 0}
+    1: {1, 5}
+    2: {2, 10}
+    3: {3, 15}
+    4: {4, 20}
+    5: {5, 25}
+
+Check out :doc:`Vectors <develop/vectors>` guide to learn more about
+Velox vectors.
+
+Expressions
+-----------
+
+Now, let’s compute a sum of 'a' and 'b' by evaluating 'a + b' expression.
+
+First we need to parse the expression into a fully typed expression tree.
+Then, we need to compile it into an executable ExprSet.
+
+.. code-block:: c++
+
+    auto exprSet = compileExpression("a + b", asRowType(data->type()));
```

Comment:
Please explicitly state here that compileExpression uses DuckDB parser for the string "a + b". The Velox expression tree syntax is a more low-level API.